streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -304,8 +304,8 @@ void handleRevocation(final Collection<TopicPartition> revokedPartitions) {
         }
 
         if (!remainingPartitions.isEmpty()) {
-            throw new IllegalStateException("Some revoked partitions that do not belong " +
-                "to any tasks remain: " + remainingPartitions);
+            log.info("The following partitions {} are missing from the task partitions. It is possible that" +
+                "they have been cleaned up by the onAssignment callback", remainingPartitions);
```

Comment:
This doesn't really make sense, `onPartitionsRevoked` is always called before `onAssignment`

Reply:
Not sure if we want to log as warn or debug though -- hitting this indicates we hit the (likely not _that_ common) race condition, or a bug.